diagnostic.md

```diff
@@ -40,31 +57,32 @@ like to show all movies favorited by a profile on
 `http://localhost:3000/profiles/1`
 
   ```md
-    # < Your Response Here >
+    # Serializers filter and reshape data.
   ```
 
   ```rb
   class ProfileSerializer < ActiveModel::Serializer
+    attributes :given_name, :surname, :email
   end
   ```
 
 1.  What would the command be to _scaffold_ out a **join table** for Favorites from
 the above `Movies` and `Profiles`.
 
   ```sh
-    # < Your Response Here >
+    # bin/rails generate scaffold favorites profile:references movie:references
   ```
 
 1.  What is `Dependent: Destroy` and where/why would we use it?
 
   ```md
-    # < Your Response Here >
+    # it destroys a favorite in order to delete a movie or profile. We would use it in the models.
```

Comment:
`dependent: :destroy` will cause the deletion of an object to also delete all associated objects.  For example, if a `User` model has `has_many :favorites, dependent: :destroy`, then deleting a particular user will also delete all rows in the `favorites` table where `user_id` refers to the `id` of the user being deleted.